src/transform3d.rs

```diff
@@ -335,6 +337,20 @@ where T: Copy + Clone +
         self.m43.approx_eq(&other.m43) && self.m44.approx_eq(&other.m44)
     }
 
+    /// Returns true is this transform is approximately equal to the other one, using
+    /// a provided epsilon value.
+    pub fn approx_eq_eps(&self, other: &Self, eps: T) -> bool
```

Comment:
I do wonder if such epsilon comparison is useful at this level. Do you have a use case in mind?
A small change in one of the components may contributing to arbitrary differences in the transformed vectors, so it's not like we can reason about the locality of error here.

Reply:
Well, this one we already have a use case for in WebRender with on of Glenn's recent patches. I don't think the change is completely arbitrary. It's as arbitrary as any similar chain of additions and multiplications where a small increment somewhere could be magnified by a factor coming from another term, but very very tiny epsilon still lead to tiny changes in practice unless you apply a *huge* scale or translation.

To extend my thinking a bit, I believe that what doesn't make sens is to have approx_eq without explicitly passing the epsilon because the threshold that you need always depends on what arithmetic you do beforehand and a bit of knowledge about reasonable input ranges for your usecase.

Reply:
It's not just about huge scaling. This is a 3D transform that is subject to perspective. Any changes to W component bring a huge deal of non-linearity to this.